Assets/Scripts/Dialog/DialogController.cs

```diff
@@ -0,0 +1,235 @@
+        {
+            if (proximalDialog != speaker.DialogReference)
+            {
+                proximalDialog = speaker.DialogReference;
+            }
+        }
+
+
+        public void ExitSpeakerZone(DialogSpeaker speaker)
+        {
+            if (proximalDialog == speaker.DialogReference)
+            {
+                proximalDialog = null;
+            }
+        }
+
+
+
+        public void Speak(string speakerTechnicalName, string text)
+        {
+            foreach (DialogSpeaker speaker in speakers)
+            {
+                if (speaker.character.GetObject().TechnicalName == speakerTechnicalName)
+                {
+                    speaker.Speak(text);
+                }
+                else
+                {
+                    speaker.ShutUp();
+                }
+            }
+        }
+
+        public void EndDialog()
+        {
+            foreach (DialogSpeaker speaker in speakers)
+            {
+                speaker.ShutUp();
+            }
+        }
+
+        public void PositionDialogBoxes()
+        {
+            //Get Average X and Y of Speakers in screen coordinates
+            Vector2 average = new Vector2();
+            foreach (DialogSpeaker speaker in speakers)
+            {
+                if (speaker.IsOnScreen())
+                {
+                    Vector2 screenPos = speaker.GetScreenPostion();
+                    average += screenPos;
+                }
+            }
+            average /= speakers.Count;
+
+            //Pick a quadrant to place the dialog box based on where the speaker is
+            //in relation to the average point
+            foreach (DialogSpeaker speaker in speakers)
+            {
+                if (speaker.GetScreenPostion().x >= average.x)
+                {
+                    if (speaker.GetScreenPostion().y >= average.y)
+                    {
+                        speaker.dialogView.mainTracker.SetOffsets(spacing, spacing);
+                    }
+                    else
+                    {
+                        speaker.dialogView.mainTracker.SetOffsets(-spacing, spacing);
+
+                    }
+                }
+                else
+                {
+                    if (speaker.GetScreenPostion().y >= average.y)
+                    {
+                        speaker.dialogView.mainTracker.SetOffsets(spacing, -spacing);
+                    }
+                    else
+                    {
+                        speaker.dialogView.mainTracker.SetOffsets(-spacing, -spacing);
+                    }
+                }
+            }
+        }
+
+        public void OnBranchesUpdated(IList<Branch> aBranches)
+        {
+
+            dialogOptions = aBranches;
+
+            if (aBranches.Count > 1)
+            {
+                dialogOptionHolder.SetActive(true);
+                awaitingResponse = true;
+                Text[] options = dialogOptionHolder.GetComponentsInChildren<Text>();
```

Comment:
This makes me a little nervous, since I don't know how consistent unity is about grabbing UI elements in the same order every time. It might be better to have a list of direct references.

Reply:
That works. Or just update the callback along with the text to point at the right branch option.